src/main/java/Duke.java

```diff
@@ -9,64 +9,117 @@ public static void main(String[] args) {
+        }
 
         while(!userInput.equals("bye")){
-            if(userInput.equals("list")){
-                System.out.println("    ____________________________________________________________");
-                System.out.println("     Here are the tasks in your list:");
-                count2 = 0;
-                count3 = 1;
-                while(count2 < count){
-                    System.out.println("     " + count3 + "." + tasks[count2].toString());
-                    count2++;
-                    count3++;
-                }
-                System.out.println("    ____________________________________________________________\n");
+            if(isProblem) {
+                isProblem = false;
 
-            } else if(userInput.equals("done 2")) {
-                System.out.println("    ____________________________________________________________");
-                System.out.println("     Nice! I've marked this task as done: ");
-                count2 = 0;
-                while(count2 < count){
-                    if(tasks[count2].getDone() == true) {
-                        System.out.println("       " + tasks[count2].getStatusIcon() + " " + tasks[count2].getDescription());
-                    }
-                    count2++;
-                }
-                System.out.println("    ____________________________________________________________");
             } else {
-                if(userInput.contains("todo")){
-                    content2 = userInput.substring(5);
-                    Task t = new Todo(content2);
-                    tasks[count] = t;
+                if (userInput.equals("list")) {
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     Here are the tasks in your list:");
+                    count2 = 0;
+                    count3 = 1;
+                    while (count2 < count) {
+                        System.out.println("     " + count3 + "." + tasks[count2].toString());
+                        count2++;
+                        count3++;
+                    }
+                    System.out.println("    ____________________________________________________________\n");
 
-                } else if(userInput.contains("deadline")){
-                    userInput = userInput.substring(9);
-                    content2 = userInput.substring(0, userInput.indexOf("/by") - 1);
-                    content3 = userInput.substring(userInput.indexOf("/by") + 4);
-                    Task t = new Deadline(content2, content3);
-                    tasks[count] = t;
+                } else if (userInput.equals("done 2")) {
+                    System.out.println("    ____________________________________________________________");
+                    System.out.println("     Nice! I've marked this task as done: ");
+                    count2 = 0;
+                    while (count2 < count) {
```

Comment:
I am not very sure what you are checking here, but you should changing the isDone in the particular task to true by calling task.markAsDone() before printing